core/web/api/links.py

```diff
@@ -66,3 +66,51 @@ def multiupdate(self):
             updated.append(link.id)
 
         return render({"updated": updated})
+
+    @route("/", methods=['POST'])
+    @requires_permissions('write')
+    def new(self):
+        """Create a new link
+
+        Create a new link from the JSON object passed in the ``POST`` data.
+
+        :<json object params: JSON object containing object ids to link
+        """
+
+        type_map = {
+            'observable': observables.Observable,
+            'entity': entities.Entity,
+            'indicator': indicators.Indicator
+        }
+
+        mandatory_params = ['type_src', 'type_dst', 'link_src', 'link_dst']
+        params = request.json
+
+        if all(key in params for key in mandatory_params):
+
+            type_src = params.get('type_src')
```

Comment:
To be extra clear, the canonical way to access values in dicts is `params['type_src']`. This will raise an exception if the key does not exist in the dict. You generally use `get` only if you want to "soft-fail".

In this case, I think using `params['type_dst']` instead of `params.get('type_dst')` is clearer with respects to your intentions: you're counting on the fact that `'type_dst'` is actually present in `params` (you even check for this in line 86, so that's great)

Using get on `type_map` is appropriate, since maybe `type_src` is denoting a type that doesn't exist, and you soft-fail and check if these are valid values or not, which is great. Just note that the default return value for `.get()` is already `None`, so specifying `.get(key, None)` is redundant. You can just get away with `.get(key)`